src/logic/safe/api/fetchTokenCurrenciesBalances.ts

```diff
@@ -20,13 +22,85 @@ type FetchTokenCurrenciesBalancesProps = {
 
 export const fetchTokenCurrenciesBalances = async ({
   safeAddress,
-  selectedCurrency,
+  selectedCurrency = 'USD',
   excludeSpamTokens = true,
-  trustedTokens = false,
 }: FetchTokenCurrenciesBalancesProps): Promise<BalanceEndpoint> => {
   const address = checksumAddress(safeAddress)
-  return getBalances(getClientGatewayUrl(), getNetworkId().toString(), address, selectedCurrency, {
-    exclude_spam: excludeSpamTokens,
-    trusted: trustedTokens,
+
+  const items = await toItems(address, selectedCurrency, excludeSpamTokens)
+
+  return {
+    fiatTotal: items.reduce((sum, token) => sum + Number(token.fiatBalance), 0).toString(),
+    items: items.sort((a, b) => {
+      return a.fiatBalance < b.fiatBalance ? 1 : -1
+    }),
+  }
+}
+const WEI_PER = 1_000_000_000_000_000_000
```

Comment:
Does this also need to be a BigNumber value?

Reply:
I don't believe so. 

Its' an integer so its precision is not a worry. and Bignumber.divide takes a plain number or a string or a big-number. 

Ive seen this pattern used in other dapps.

Reply:
👍